tasks/task_database.c

```diff
@@ -474,14 +533,28 @@ static int task_database_iterate_playlist(
             return cue_get_crc(name, &db_state->crc);
          }
          break;
+      case FILE_TYPE_GDI:
+         db_state->serial[0] = '\0';
+         if (gdi_get_serial(name, db_state->serial))
+         {
+            database_info_set_type(db, DATABASE_TYPE_SERIAL_LOOKUP);
+         }
+         else
+         {
+            database_info_set_type(db, DATABASE_TYPE_CRC_LOOKUP);
+            return gdi_get_crc(name, &db_state->crc);
+         }
+         break;
       case FILE_TYPE_ISO:
          db_state->serial[0] = '\0';
          file_get_serial(name, 0, SIZE_MAX, db_state->serial);
          database_info_set_type(db, DATABASE_TYPE_SERIAL_LOOKUP);
          break;
       case FILE_TYPE_CHD:
          db_state->serial[0] = '\0';
-         if (chd_get_serial(name, db_state->serial))
+         /* There are no serial databases, so don't bother with
+            serials at the moment */
+         if (0 && chd_get_serial(name, db_state->serial))
```

Comment:
This will always skip? Should we just remove this code?

Reply:
It prevents an unused function warning, and make it easy to enable later.